### PR TITLE
[DONOTMERGE]Fix FP8 wo_a GEMM precision on Blackwell: quantize with UE8M0 scale

### PR DESCRIPTION
**Pull request by yhyang201**
## Summary

Fix a precision issue in DeepSeek-V4's `wo_a` FP8 GEMM path that caused EAGLE speculative decoding accept_length to drop from 2.76 (H200) to 2.69 (B300/B200).

This PR includes #26766 (fuse UE8M0 rounding into quant kernel) as the first commit, and adds a precision fix on top.

## Root Cause

The `wo_a` FP8 path quantized activations with an exact FP32 scale, then rounded the scale to UE8M0 (power-of-2) *after* quantization:

```python
# Before (wrong order):
o_fp8, o_s = quant_fp8(o, group_size=128)   # quantize with FP32 scale
o_s = ceil_to_ue8m0(o_s)                     # round scale AFTER quantization
fp8_einsum(o_fp8, o_s, weight)               # GEMM uses rounded scale
# -> FP8 values quantized with one scale, GEMM dequants with a different scale
```

Measured impact of the scale mismatch:
- Scale inflation ratio (ceil/original): **mean=1.31, max=1.99**
- Quantization error amplification: **14.1x** vs using consistent scale
- Average relative error from rounding: **31%**

### Why Blackwell is affected more than Hopper

On **Hopper (SM90)**, `DEEPGEMM_SCALE_UE8M0=False` -- DeepGEMM processes scales as FP32 post-GEMM corrections, partially masking the mismatch.

On **Blackwell (SM100)**, `DEEPGEMM_SCALE_UE8M0=True` -- DeepGEMM packs scales into native UE8M0 format for hardware tensor-core scaling, **fully exposing the mismatch**. The `wo_a` GEMM runs in every attention layer (62 layers for DSV4-Flash), and the error accumulates through residual connections.

### The correct approach (matches official DeepSeek-V4 kernel.py)

The [official DeepSeek-V4 inference kernel](https://huggingface.co/deepseek-ai/DeepSeek-V4-Pro/blob/main/inference/kernel.py) computes the UE8M0 scale *first*, then quantizes with it:

```python
# Official kernel.py act_quant(scale_fmt="ue8m0"):
scale = fast_round_scale(amax / fp8_max)    # round to power-of-2 FIRST
o_fp8 = round_to_fp8(o / scale)              # quantize WITH rounded scale
# -> quantization and dequantization use the SAME scale
```

## Minimal fix (without fused kernel)

For anyone who wants to apply just the precision fix without the fused kernel from #26766, the following pure-PyTorch patch on `deepseek_v4.py` is sufficient:

```diff
         if _FP8_WO_A_GEMM:
             import deep_gemm

             T, G, D = o.shape
             R = self.o_lora_rank
-            o_fp8, o_s = sglang_per_token_group_quant_fp8(
-                o.reshape(T * G, D).contiguous(),
-                group_size=128,
-            )
-            o_s = deep_gemm.ceil_to_ue8m0(o_s)
+            # Compute UE8M0 scale first, then quantize with it
+            o_flat = o.reshape(T * G, D).contiguous()
+            _group_size = 128
+            o_groups = o_flat.float().view(-1, D // _group_size, _group_size)
+            amax = o_groups.abs().amax(dim=-1, keepdim=True).clamp(min=1e-12)
+            _fp8_max = torch.finfo(torch.float8_e4m3fn).max
+            o_s = torch.pow(2.0, torch.ceil(torch.log2(amax / _fp8_max)))
+            o_fp8 = (o_groups / o_s).clamp(-_fp8_max, _fp8_max).to(torch.float8_e4m3fn)
+            o_fp8 = o_fp8.view(T * G, D)
+            o_s = o_s.view(T * G, D // _group_size)
```

This PyTorch version was tested and produces the same accept_len improvement (B300: 2.69 -> 2.76), but launches multiple small kernels instead of one fused kernel.

## Fix in this PR (fused kernel)

The fused CUDA kernel in `per_token_group_quant_8bit_v2.cuh` had a special `SCALE_UE8M0 && !IS_COLUMN_MAJOR` branch that quantized with the exact scale and only rounded the output scale. This PR removes that branch and unifies all UE8M0 paths to use `calculate_fp8_scales<SCALE_UE8M0=true>`, which computes the rounded scale first and then quantizes with it -- single fused kernel, correct precision.

## Results

**Pure PyTorch fix (no fused kernel):**
| Platform | Before | After | Delta |
|----------|--------|-------|-------|
| B300 (SM100, trtllm, CI default config) | 2.69 | **2.76** | +0.07 |
| B300 (SM100, MegaMoE + marlin dense) | 2.71 | **2.79** | +0.08 |

**Fused CUDA kernel (this PR):**
| Platform | Before | After | Delta |
|----------|--------|-------|-------|
| H200 (SM90, marlin) | 2.76 | **2.76** | unchanged |
| B300 (SM100, trtllm, CI default config) | 2.69 | **2.77** | **+0.08** |

All 9 CI tests pass on both H200 and B300.

### CODE REVIEW

**Comment by yhyang201** [comment]
Merged via PR #26766.
